p4-16/spec/P4-16-spec.mdk

```diff
@@ -4111,6 +4111,17 @@ expressions are legal:
   [#sec-uninitialized-values-and-writing-invalid-headers] for more
   details.
 
+  The `index` is an expression that must be one of the following types:
+
+  - `int` - an infinite-precision integer (section
+    [#sec-infinite-precision-integers])
+  - `bit<W>` - a `W`-bit unsigned integer where `W >= 1` (section
+    [#sec-unsigned-integers])
+  - `int<W>` - a `W`-bit signed integer where `W >= 1` (section
+    [#sec-signed-integers])
+  - a serializable `enum` with an underlying type that is `bit<W>` or
+    `int<W>` (section [#sec-enum-types]).
```

Comment:
@jafingerhut AI Andy to create a small test program and see whether p4c accepts a header stack index with type of serializable enum is supported today.

Reply:
A simple test program shows that recent versions of p4c given an error when one attempts to use an expression of type `enum bit<8> ...` as the index of a header stack reference: https://github.com/p4lang/p4c/issues/2726